### PR TITLE
Handle Mouse events rather than polling

### PR DESCRIPTION
**Pull request by peppy**
Need people to test this and see how it performs on different operating systems. Also whether it is more or less RAW than previous.

- Need macOS users to test if this fixes cursor input not working for some devices.
- Need windows users to test whether this has acceleration or not.
- Need linux users to test how this works in general.

As I'm currently running windows in a virtualised device, I haven't tested native behaviour yet myself. OpenTK docs suggest using the events gives raw input response, but from my testing so far this doesn't seem correct.

### CODE REVIEW

**Comment by peppy** [comment]
Tested with native mouse on windows, does not bypass acceleration. See #775 for a different attempt.

**Comment by peppy** [comment]
Continuing developing this PR as a base implementation (replacement for previous non-raw input polling handler).

**Comment by smoogipooo** [comment]
So do we still want to merge this one in or look towards merging https://github.com/ppy/osu-framework/pull/775 ? This PR looks fine (also, doesn't bypass acceleration for me either).

File: osu.Framework.Desktop/Input/Handlers/Mouse/OpenTKMouseHandler.cs [hunk 1]
```diff
@@ -5,46 +5,72 @@
...
+        {
+            base.Dispose(disposing);
+            scheduled.Cancel();
+        }
 
-                Point point = host.Window.PointToClient(new Point(state.X, state.Y));
-                Vector2 pos = new Vector2(point.X, point.Y);
+        private void handleMouseEvent(MouseEventArgs e)
+        {
+            if (!host.Window.Visible || !mouseInWindow)
```
Comment (smoogipooo) [inline osu.Framework.Desktop/Input/Handlers/Mouse/OpenTKMouseHandler.cs]: Is this conditional required? As far as I can tell it's never being triggered for me, different platform implementation differences?
Am fine with keeping this around but if there is such a difference it should probably be documented.

Comment (peppy) [inline osu.Framework.Desktop/Input/Handlers/Mouse/OpenTKMouseHandler.cs]: I believe there's a change GetCursorState is called before the window handle is fully created, PointInClient will exception in such cases

Comment (smoogipooo) [inline osu.Framework.Desktop/Input/Handlers/Mouse/OpenTKMouseHandler.cs]: Aha, triggered it.

Comment (smoogipooo) [inline osu.Framework.Desktop/Input/Handlers/Mouse/OpenTKMouseHandler.cs]: Wait but this path hasn't anything to do with PointInClient. Shouldn't this visibility check be added t to the GetCursorState path?

Comment (peppy) [inline osu.Framework.Desktop/Input/Handlers/Mouse/OpenTKMouseHandler.cs]: yep, this was moved around incorrectly

**Pull request merged**